src/Sylius/Bundle/ApiBundle/Command/Checkout/ChoosePaymentMethod.php

```diff
@@ -15,64 +15,30 @@
 
 use Sylius\Bundle\ApiBundle\Command\IriToIdentifierConversionAwareInterface;
 use Sylius\Bundle\ApiBundle\Command\OrderTokenValueAwareInterface;
+use Sylius\Bundle\ApiBundle\Command\PaymentIdAwareInterface;
 use Sylius\Bundle\ApiBundle\Command\PaymentMethodCodeAwareInterface;
 
-class ChoosePaymentMethod implements OrderTokenValueAwareInterface, PaymentMethodCodeAwareInterface, IriToIdentifierConversionAwareInterface
+class ChoosePaymentMethod implements OrderTokenValueAwareInterface, PaymentIdAwareInterface, PaymentMethodCodeAwareInterface, IriToIdentifierConversionAwareInterface
 {
-    /** @var string|null */
-    public $orderTokenValue;
-
-    /**
-     * @immutable
-     *
-     * @var string|null
-     */
-    public $paymentId;
-
-    /**
-     * @immutable
-     *
-     * @var string|null
-     */
-    public $paymentMethodCode;
-
-    public function __construct(string $paymentMethodCode)
-    {
-        $this->paymentMethodCode = $paymentMethodCode;
+    public function __construct(
+        public string $paymentMethodCode,
+        public ?int $paymentId = null,
+        public ?string $orderTokenValue = null,
+    ) {
```

Comment:
```suggestion
    public function __construct(
        public string $orderTokenValue,
        public int $paymentId,
        public string $paymentMethodCode,
    ) {
```
1. IMO it's a more natural order.
2. Should the command now be strictly typed since we create them with all the necessary data?

Reply:
In your example, it's unclear which argument is from the body. I also think it would be better to have a default value for the argument which will be set by the context builder.